django/utils/http.py

```diff
@@ -297,7 +297,8 @@ def is_safe_url(url, allowed_hosts=None, require_https=False):
     if not url:
         return False
     if allowed_hosts is None:
-        allowed_hosts = set()
+        from django.conf import settings
```

Comment:
I believe you can move this import to the top of the module. Doing so does not cause any tests to fail for me.